src/HDF5.jl

```diff
@@ -922,7 +922,7 @@ end
 exists(attr::Attributes, path::String) = h5a_exists(checkvalid(attr.parent).id, path)
 exists(dset::Union{Dataset,Datatype}, path::String) = h5a_exists(checkvalid(dset).id, path)
 exists(parent::Union{File,Group}, path::String) = exists(parent, path, Properties())
-has(parent::Union{File,Group,Dataset}, path::String) = exists(parent, path)
+Base.haskey(parent::Union{File,Group,Dataset}, path::String) = exists(parent, path)
```

Comment:
```suggestion
Base.haskey(parent::Union{File,Object,Attributes}, path::String) = exists(parent, path)
```
Looks like all those types are covered by `exists`, though maybe `exists` should also be deprecated in favor of just `haskey`, the existing methods renamed, and this one removed.

Reply:
I was wondering about that. Good to know you are also thinking the same. 

This also aligns closer with the `dict` like interface we want to transition towards.

Reply:
I'm going to go ahead with this bigger change, might as well have it in a single coherent PR.